modules/apps/remote-app/remote-app-web/src/main/java/com/liferay/remote/app/web/internal/deployer/RemoteAppEntryDeployerImpl.java

```diff
@@ -68,8 +69,22 @@ private String _getPortletCategoryName(RemoteAppEntry remoteAppEntry) {
 	}
 
 	private String _getPortletId(RemoteAppEntry remoteAppEntry) {
-		return "com_liferay_remote_app_web_internal_portlet_" +
-			"RemoteAppEntryPortlet_" + remoteAppEntry.getRemoteAppEntryId();
+		String portletId = remoteAppEntry.getPortletAlias();
```

Comment:
Shouldn't we also add a validation to make sure `portletAlias`es are unique?

Otherwise, if they collide, the registration will fail and it may render the whole feature unusable when the portal is started...

Also note that should be unique for ANY company. This can create a mismatch and/or unexpected validation failures because `RemoteAppEntry`s are scoped by company so, for example:

1. Customer `A` creates a remote app `X` and configures its `portletAlias` as `perico`
2. Customer `B` tries to use `perico` as well for his remote app `Y`, but the portal doesn't allow him to do that to avoid collisions.

In this scenario, customer `B` would be quite puzzled, because he can't even see the portlet `X` from customer `A` in the dataset :grimacing: